packages/core/src/components/forms/_file-upload.scss

```diff
@@ -69,20 +70,20 @@ $file-upload-button-width-large: $pt-grid-size * 8.5;
   top: 0;
   right: 0;
   left: 0;
-  margin-right: $file-upload-button-width;
   border-top-right-radius: 0;
   border-bottom-right-radius: 0;
   height: $pt-input-height;
+  padding-right: $file-upload-button-width + $input-padding-horizontal;
   line-height: $pt-input-height;
   color: $pt-text-color-muted;
   user-select: none;
 
+  // ::after so that it's relatively easy to add an icon with ::before
   &::after {
     @include pt-button();
     position: absolute;
     top: 0;
     right: 0;
-    left: 100%;
     margin-left: -$button-border-width;
     border-radius: $pt-border-radius;
     border-top-left-radius: 0;
```

Comment:
Yeah, sorry, missed this one and couldn't actually tell what the problem was until I saw @cmslewis's GIF. Thanks for that! Would never catch something that subtle on my own.